elfo-core/src/supervisor/backoff.rs

```diff
@@ -1,20 +1,20 @@
 use std::time::Duration;
 
+use crate::RestartParams;
 use elfo_utils::time::Instant;
 
-const BACKOFF_STEP: Duration = Duration::from_secs(5);
-const MAX_BACKOFF: Duration = Duration::from_secs(30);
-
 pub(crate) struct Backoff {
-    next_backoff: Duration,
+    next_backoff: Option<Duration>,
     start_time: Instant,
+    restart_count: u64,
 }
 
 impl Default for Backoff {
     fn default() -> Self {
         Self {
-            next_backoff: BACKOFF_STEP,
+            next_backoff: None,
```

Comment:
It seems it's the only place where it can become `None` =(

Reply:
I think it's better to remove `next_backoff` totally (his function is duplicated by `restart_count`)

Reply:
I simplified a bit backoff.rs